interactive_engine/compiler/src/test/java/com/alibaba/graphscope/common/ir/ExpressionTest.java

```diff
@@ -0,0 +1,168 @@
+    }
+
+    @Test
+    public void variable_1_test() {
+        RexNode node = builder.source(mockSourceConfig(null)).variable((String) null);
+        Assert.assertEquals("DEFAULT", node.toString());
+    }
+
+    @Test
+    public void variable_2_test() {
+        RexNode node = builder.source(mockSourceConfig("a")).variable("a");
+        Assert.assertEquals("a", node.toString());
+    }
+
+    @Test
+    public void variable_3_test() {
+        try {
+            builder.source(mockSourceConfig(null)).variable("a");
+        } catch (IllegalArgumentException e) {
+            return;
+        }
+        Assert.fail("tag 'a' should not exist");
+    }
+
+    @Test
+    public void variable_4_test() {
+        RexNode node = builder.source(mockSourceConfig(null)).variable(null, "name");
+        Assert.assertEquals(node.getType().getSqlTypeName(), SqlTypeName.CHAR);
+        Assert.assertEquals("DEFAULT.name", node.toString());
+    }
+
+    @Test
+    public void variable_5_test() {
+        RexNode node = builder.source(mockSourceConfig("a")).variable("a", "age");
+        Assert.assertEquals(node.getType().getSqlTypeName(), SqlTypeName.INTEGER);
+        Assert.assertEquals("a.age", node.toString());
+    }
+
+    // a.age + 10
+    @Test
+    public void plus_test() {
+        RexNode var = builder.source(mockSourceConfig("a")).variable("a", "age");
+        RexNode plus = builder.call(GraphStdOperatorTable.PLUS, var, builder.literal(10));
+        Assert.assertEquals(plus.getType().getSqlTypeName(), SqlTypeName.INTEGER);
+        Assert.assertEquals("+(a.age, 10)", plus.toString());
+    }
+
+    // a.age == 10
+    @Test
+    public void equal_1_test() {
+        RexNode var = builder.source(mockSourceConfig("a")).variable("a", "age");
+        RexNode equal = builder.call(GraphStdOperatorTable.EQUALS, var, builder.literal(10));
+        Assert.assertEquals(equal.getType().getSqlTypeName(), SqlTypeName.BOOLEAN);
+        Assert.assertEquals("=(a.age, 10)", equal.toString());
+    }
+
+    // a.age == 'X'
+    // Integer is comparable with String in Calcite standard implementation
+    // todo: maybe we need rewrite the implementation to add more constraints
+    @Test
+    public void equal_2_test() {
+        RexNode var = builder.source(mockSourceConfig("a")).variable("a", "age");
+        RexNode equal = builder.call(GraphStdOperatorTable.EQUALS, var, builder.literal("X"));
+        Assert.assertEquals(equal.getType().getSqlTypeName(), SqlTypeName.BOOLEAN);
+        Assert.assertEquals("=(a.age, 'X')", equal.toString());
+    }
+
+    // a.age + 10 > 30
+    @Test
+    public void equal_plus_test() {
+        RexNode var = builder.source(mockSourceConfig("a")).variable("a", "age");
+        RexNode plus = builder.call(GraphStdOperatorTable.PLUS, var, builder.literal(10));
+        RexNode equal = builder.call(GraphStdOperatorTable.EQUALS, plus, builder.literal(30));
```

Comment:
According to your comment, this should be `GreaterThan` ?

Reply:
fix comments